### ISSUE TITLE
E11000 duplicate key error collection when migrate credentials collection

### ISSUE THREAD

**Issue by gongzhao1** [eclipse-hono/hono-extras#9]
Hi,
**Version Information**
old hono version: 1.2.3
upgrade hono version: 1.6.0

**Environment Information**
Kubernetes Cluster: 1.15.6
Mongodb version:  3.6.7 replicaset mode

**Description**
I try to migrate my data from file-based to Mongodb. I follow the README instruction to do the migration, Unfortunately, when I try to import credentials collection into Mongodb, below error occurs.
```
mongodb@mongodb-replicaset-0:/work-dir/test$ mongoimport -d DEFAULT_DB -c credentials --jsonArray --file credentials-converted.json
2021-03-17T03:09:08.888+0000    connected to: localhost
2021-03-17T03:09:09.100+0000    num failures: 6
2021-03-17T03:09:09.100+0000    error inserting documents: multiple errors in bulk operation:
  - E11000 duplicate key error collection: DEFAULT_DB.credentials index: tenant-id_1_device-id_1 dup key: { : "***", : "device01" }
  - E11000 duplicate key error collection: DEFAULT_DB.credentials index: tenant-id_1_device-id_1 dup key: { : "***", : "com.lab.iot:device01" }
  - E11000 duplicate key error collection: DEFAULT_DB.credentials index: tenant-id_1_device-id_1 dup key: { : "***", : "device01" }
  - E11000 duplicate key error collection: DEFAULT_DB.credentials index: tenant-id_1_device-id_1 dup key: { : "***", : "com.test:gateway001" }

2021-03-17T03:09:09.100+0000    imported 870 documents
```

**Comment by gongzhao1** [comment]
I find there are some index keys in credentials collection.
```
[
	...
    {
            "v" : 2,
            "unique" : true,
            "key" : {
                    "tenant-id" : 1,
                    "device-id" : 1
            },
            "name" : "tenant-id_1_device-id_1",
            "ns" : "DEFAULT_DB.credentials"
    },
	...
]
```
So I check my converted-credentials.json, found some error records
```
[
    {
        "tenant-id": "***",
        "device-id": "com.lab.iot:device01",
        "version": "d2d6a829-36b3-4532-a753-01ef46e43f26",
        "updatedOn": "2021-03-17T15:10:19Z",
        "credentials": [
            {
                "auth-id": "com.lab.iot:device01",
                "type": "hashed-password",
                "secrets": [
                    {
                        "hash-function": "bcrypt",
                        "pwd-hash": "$2a$10$4.G8GSen4tntN1plEbJYjO8PfDo8JTjDWHPh5S4JUcMvWBz7DwMaS",
                        "id": "1336a835-0eb8-4813-a769-96c825f064bc"
                    }
                ],
                "enabled": true
            }
        ]
    },
    {
        "tenant-id": "***",
        "device-id": "com.lab.iot:device01",
        "version": "5333222c-a527-4f61-91f5-31d5b6a6650c",
        "updatedOn": "2021-03-17T15:10:19Z",
        "credentials": [
            {
                "auth-id": "device01",
                "type": "hashed-password",
                "secrets": [
                    {
                        "hash-function": "bcrypt",
                        "pwd-hash": "$2a$10$OctvOO0PGreKMGAXQagPLumw/i8qK2iLXR5BT2dj4knARjPb6hxh6",
                        "id": "a0925a69-a3b5-4142-aa24-23186064f123"
                    }
                ],
                "enabled": true
            }
        ]
    }
]
```
**The above records tenant-id and device-id are the same, so their credentials should be merged together**, The correct record should be as follows
```
[
    {
        "tenant-id": "***",
        "device-id": "com.lab.iot:device01",
        "version": "d2d6a829-36b3-4532-a753-01ef46e43f26",
        "updatedOn": "2021-03-17T15:10:19Z",
        "credentials": [
            {
                "auth-id": "com.lab.iot:device01",
                "type": "hashed-password",
                "secrets": [
                    {
                        "hash-function": "bcrypt",
                        "pwd-hash": "$2a$10$4.G8GSen4tntN1plEbJYjO8PfDo8JTjDWHPh5S4JUcMvWBz7DwMaS",
                        "id": "1336a835-0eb8-4813-a769-96c825f064bc"
                    }
                ],
                "enabled": true
            },
			{
                "auth-id": "device01",
                "type": "hashed-password",
                "secrets": [
                    {
                        "hash-function": "bcrypt",
                        "pwd-hash": "$2a$10$OctvOO0PGreKMGAXQagPLumw/i8qK2iLXR5BT2dj4knARjPb6hxh6",
                        "id": "a0925a69-a3b5-4142-aa24-23186064f123"
                    }
                ],
                "enabled": true
            }
        ]
    }
]
```
So something is wrong with the `_transform_credentials` function.

**Comment by gongzhao1** [comment]
I modified this function `_transform_credentials` and passed the test in my environment.
The corresponding PR will be submitted

**Comment by kaniyan** [comment]
@gongzhao1 Thanks for reporting the issue and contributing a PR for this.

This has been closed via PR #11.
